src/include/sof/audio/component.h

```diff
@@ -527,6 +527,24 @@ static inline struct sof_ipc_comp_config *dev_comp_config(struct comp_dev *dev)
 	return (struct sof_ipc_comp_config *)(&dev->comp + 1);
 }
 
+/**
+ * Allocates memory for the component device and initializes common part.
+ * @param drv Parent component driver.
+ * @param bytes Size of the component device in bytes.
+ * @return Pointer to the component device.
+ */
+static inline struct comp_dev *comp_alloc(const struct comp_driver *drv,
+					  size_t bytes)
+{
+	struct comp_dev *dev = NULL;
+
+	dev = rzalloc(SOF_MEM_ZONE_RUNTIME, 0, SOF_MEM_CAPS_RAM, bytes);
```

Comment:
@mmaka1 in theory rzalloc can fail and dev could be NULL no? shoudnt we check for that?

Reply:
Of course, there should be a check before accessing the object, fixed.